reactioncommerce/reaction
# Fix: 404 on Hydra Oauth page

Resolves #4794
Impact: **major**
Type: **bugfix**

## Issue Description
OAuth IDP login pages created by the Hydra plugin are not showing up in the latest release candidate 6. So `/account/login` returns Not Found. Going back a few commits (e.g `d37e12b2ea92`), the page shows up fine.

## Solution
Calling `Reaction.addRolesToGroups()` to add the "account/login" permission to guest & customer groups was the first step. This was called in an `afterCoreInit` event callback in the hydra-oauth core plugin (see @impactmass's original PR - https://github.com/reactioncommerce/reaction/pull/4795/files)

Even with that, it was taking 2 server restarts before the account/login view was working (not returning a 404). I traced this down to Migration 5 - which was deleting and recreating all groups. Because we now have a core `createGroups` function that does this before the migrations run, this migration is no longer needed.



## Breaking changes
None

## Testing
1. Run `reaction reset -n`
2. Startup Reaction via the platform and confirm you can login and signup through clicking the buttons in the starterkit
3. Confirm you can still login as admin in Reaction and have correct permissions (update shop info, enable payment & shipping methods, etc)
4. Confirm you can still checkout in Starterkit & Reaction
